modules/gapi/src/executor/gtbbexecutor.cpp

```diff
@@ -6,7 +6,7 @@
 
 #include "gtbbexecutor.hpp"
 
-#if defined(HAVE_TBB)
+#if defined(HAVE_TBB) && !defined(OPENCV_TBB_USE_VERSION_H)
```

Comment:
Need to include `tbb.h` / `tbb/task.h` and then check `TBB_INTERFACE_VERSION < XYZ` instead.

- https://github.com/oneapi-src/oneTBB/blame/v2020.3/include/tbb/tbb_stddef.h#L25

Perhaps `XYZ` should be `< 12000`, because `-DWITH_TBB=ON -DBUILD_TBB=ON` uses 2020.2 with `interface 11102` and gapi builds succesfully.

---

BTW, Please add a comment which API is not available.

Reply:
Done.